src/contracts/misc/GhoAaveSteward.sol

```diff
@@ -0,0 +1,353 @@
+   * @param baseVariableBorrowRate The new base variable borrow rate
+   * @param variableRateSlope1 The new variable rate slope 1
+   * @param variableRateSlope2 The new variable rate slope 2
+   */
+  function _updateRates(
+    uint256 optimalUsageRatio,
+    uint256 baseVariableBorrowRate,
+    uint256 variableRateSlope1,
+    uint256 variableRateSlope2
+  ) internal {
+    IDefaultInterestRateStrategyV2.InterestRateData
+      memory rateParams = IDefaultInterestRateStrategyV2.InterestRateData({
+        optimalUsageRatio: uint16(optimalUsageRatio),
+        baseVariableBorrowRate: uint32(baseVariableBorrowRate),
+        variableRateSlope1: uint32(variableRateSlope1),
+        variableRateSlope2: uint32(variableRateSlope2)
+      });
+
+    IPoolConfigurator(IPoolAddressesProvider(POOL_ADDRESSES_PROVIDER).getPoolConfigurator())
+      .setReserveInterestRateData(GHO_TOKEN, abi.encode(rateParams));
+  }
+
+  /**
+   * @notice method to validate the interest rates update
+   * @param optimalUsageRatio The new optimal usage ratio to validate
+   * @param baseVariableBorrowRate The new base variable borrow rate to validate
+   * @param variableRateSlope1 The new variable rate slope 1 to validate
+   * @param variableRateSlope2 The new variable rate slope 2 to validate
+   */
+  function _validateRatesUpdate(
+    uint256 optimalUsageRatio,
+    uint256 baseVariableBorrowRate,
+    uint256 variableRateSlope1,
+    uint256 variableRateSlope2
+  ) internal view {
+    (
+      uint256 currentOptimalUsageRatio,
+      uint256 currentBaseVariableBorrowRate,
+      uint256 currentVariableRateSlope1,
+      uint256 currentVariableRateSlope2
+    ) = _getInterestRatesForAsset(GHO_TOKEN);
+
+    require(
+      optimalUsageRatio != currentOptimalUsageRatio ||
+        baseVariableBorrowRate != currentBaseVariableBorrowRate ||
+        variableRateSlope1 != currentVariableRateSlope1 ||
+        variableRateSlope2 != currentVariableRateSlope2,
+      'NO_CHANGE_IN_RATES'
+    );
+
+    require(
+      _updateWithinAllowedRange(
+        currentOptimalUsageRatio,
+        optimalUsageRatio,
+        _borrowRateConfig.optimalUsageRatioMaxChange,
+        false
+      ),
+      'INVALID_OPTIMAL_USAGE_RATIO'
+    );
+    require(
+      _updateWithinAllowedRange(
+        currentBaseVariableBorrowRate,
+        baseVariableBorrowRate,
+        _borrowRateConfig.baseVariableBorrowRateMaxChange,
+        false
+      ),
+      'INVALID_BORROW_RATE_UPDATE'
+    );
+    require(
+      _updateWithinAllowedRange(
+        currentVariableRateSlope1,
+        variableRateSlope1,
+        _borrowRateConfig.variableRateSlope1MaxChange,
+        false
+      ),
+      'INVALID_VARIABLE_RATE_SLOPE1'
+    );
+    require(
+      _updateWithinAllowedRange(
+        currentVariableRateSlope2,
+        variableRateSlope2,
+        _borrowRateConfig.variableRateSlope2MaxChange,
+        false
+      ),
+      'INVALID_VARIABLE_RATE_SLOPE2'
+    );
+
+    require(
```

Comment:
This is not needed anymore, since we are imposing a condition on every param.
`GHO_BORROW_RATE_MAX` can be removed too.

Reply:
the condition on the other params is the max change allowed (in one change). `GHO_BORROW_RATE_MAX` is an absolute maximum value, and is part of the spec

Reply:
let's wait for the spec confirmation

Reply:
it's confirmed that there is 2 conditions for GHO borrow rate changes: maximum total borrow rate and maximum change per parameter.